File: utc/2026-2027/Toan_roi_rac-1-1-26(N06)/tu-chup/luyen-tap/tuan-1/Bang_chan_tri.cpp
#include <iostream>
#include <iomanip>
using namespace std;

// Phu dinh
bool NOT(bool p) {
    return !p;
}

// P va Q
bool AND(bool p, bool q) {
    return p && q;
}

// P hoac Q
bool OR(bool p, bool q) {
    return p || q;
}

// P XOR Q
bool XOR(bool p, bool q) {
    return p != q;
}

// P suy ra Q: P -> Q
bool IMPLIES(bool p, bool q) {
    return !p || q;
}

// P tuong duong Q: P <-> Q
bool IFF(bool p, bool q) {
    return p == q;
}

int main() {
    cout << boolalpha;

    cout << "=====================================================================\n";
    cout << " p  q | !p | !q | p^q | p+q | p xor q | p->q | p<->q\n";
    cout << "=====================================================================\n";

    bool values[] = {true, false};

    for (bool p : values) {
        for (bool q : values) {

            cout << setw(2) << p << " "
                 << setw(2) << q << " | "
                 << setw(2) << NOT(p) << "  | "
                 << setw(2) << NOT(q) << "  | "
                 << setw(3) << AND(p, q) << " | "
                 << setw(3) << OR(p, q) << "  | "
                 << setw(7) << XOR(p, q) << " | "
                 << setw(4) << IMPLIES(p, q) << "  | "
                 << setw(5) << IFF(p, q)
                 << endl;
        }
    }

    cout << "=====================================================================\n";

    return 0;
}

File: utc/2026-2027/Toan_roi_rac-1-1-26(N06)/tu-chup/luyen-tap/tuan-1/Bang_chan_tri.py
def NOT(p):
    return not p


def AND(p, q):
    return p and q


def OR(p, q):
    return p or q


def XOR(p, q):
    return p != q


def IMPLIES(p, q):
    return (not p) or q


def IFF(p, q):
    return p == q


print("=" * 75)
print(" p  q | !p | !q | p^q | p+q | p xor q | p->q | p<->q")
print("=" * 75)

values = [True, False]

for p in values:
    for q in values:
        print(
            f"{str(p):5} "
            f"{str(q):5} | "
            f"{str(NOT(p)):5} | "
            f"{str(NOT(q)):5} | "
            f"{str(AND(p, q)):5} | "
            f"{str(OR(p, q)):5} | "
            f"{str(XOR(p, q)):7} | "
            f"{str(IMPLIES(p, q)):5} | "
            f"{str(IFF(p, q)):5}"
        )

print("=" * 75)